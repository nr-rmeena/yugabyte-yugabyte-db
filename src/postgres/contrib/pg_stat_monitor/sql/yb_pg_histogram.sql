CREATE EXTENSION pg_stat_monitor;

CREATE TABLE t1(a int);

SELECT pg_stat_monitor_reset();

INSERT INTO t1 VALUES(generate_series(1,10));
ANALYZE t1;
SELECT count(*) FROM t1;

INSERT INTO t1 VALUES(generate_series(1,10000));
ANALYZE t1;
SELECT count(*) FROM t1;;

SELECT query, calls FROM pg_stat_monitor ORDER BY query COLLATE "C";
SELECT * FROM histogram(0, 'F44CD1B4B33A47AF') AS a(range TEXT, freq INT, bar TEXT);

DROP TABLE t1;
SELECT pg_stat_monitor_reset();
DROP EXTENSION pg_stat_monitor;
