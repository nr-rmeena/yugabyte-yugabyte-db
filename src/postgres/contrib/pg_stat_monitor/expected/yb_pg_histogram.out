CREATE EXTENSION pg_stat_monitor;
CREATE TABLE t1(a int);
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

INSERT INTO t1 VALUES(generate_series(1,10));
ANALYZE t1;
SELECT count(*) FROM t1;
 count
-------
    10
(1 row)

INSERT INTO t1 VALUES(generate_series(1,10000));
ANALYZE t1;
SELECT count(*) FROM t1;;
 count
-------
 10010
(1 row)

SELECT query, calls FROM pg_stat_monitor ORDER BY query COLLATE "C";
                                query                                 | calls
----------------------------------------------------------------------+-------
 ANALYZE t1;                                                          |     2
 INSERT INTO t1 VALUES(generate_series($1,$2))                        |     2
 SELECT count(*) FROM t1;                                             |     2
 SELECT pg_stat_monitor_reset();                                      |     1
 SELECT query, calls FROM pg_stat_monitor ORDER BY query COLLATE "C"; |     1
(5 rows)

SELECT * FROM histogram(0, 'F44CD1B4B33A47AF') AS a(range TEXT, freq INT, bar TEXT);
 range | freq | bar
-------+------+-----
(0 rows)

DROP TABLE t1;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

DROP EXTENSION pg_stat_monitor;
